core/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -110,6 +114,7 @@ private[spark] abstract class YarnSchedulerBackend(
         addWebUIFilter(filterName, filterParams, proxyBase)
 
       case RemoveExecutor(executorId, reason) =>
+        logWarning(reason)
```

Comment:
Ah. I guess this is a side-effect of the race between the disconnect and the AM message. Because there's this code in `TaskSchedulerImpl`:

    override def executorLost(executorId: String, reason: ExecutorLossReason): Unit = {
      var failedExecutor: Option[String] = None
  
      synchronized {
        if (activeExecutorIds.contains(executorId)) {
          val hostPort = executorIdToHost(executorId)
          logError("Lost executor %s on %s: %s".format(executorId, hostPort, reason))